workers/miscale.py

```diff
@@ -16,14 +19,58 @@
 class MiscaleWorker(BaseWorker):
 
     SCAN_TIMEOUT = 5
-
+    
+    def getAge(self, d1):
+        d1 = datetime.strptime(str(d1), "%Y-%m-%d")
+        d2 = datetime.strptime(datetime.today().strftime('%Y-%m-%d'), '%Y-%m-%d')
+        return abs((d2 - d1).days)/365
+    
+    def _setup(self):
+        try:
+            if self.users:
+                self.is_users = True
+        except:
+            self.is_users = False
+    
     def status_update(self):
         results = self._get_data()
+        
         messages = [MqttMessage(topic=self.format_topic("weight/" + results.unit), payload=results.weight)]
         if results.impedance:
             messages.append(MqttMessage(topic=self.format_topic("impedance"), payload=results.impedance))
         if results.midatetime:
             messages.append(MqttMessage(topic=self.format_topic("midatetime"), payload=results.midatetime))
+        
+        if self.is_users:
+            for key, item in self.users.items():
+                greater = float(item['weight_template'].split(' to ')[0])
+                less = float(item['weight_template'].split(' to ')[1])
```

Comment:
Here you should be able to check using https://www.w3schools.com/python/ref_func_hasattr.asp.

Reply:
Okay

Reply:
It should be hasattr(self, 'users') right?

Reply:
Or use MiscaleWorker as object?

Reply:
Not sure probably self :D I'm not experienced Python programmer, only learning.